src/pages/dogs/DogDetailsPage.tsx

```diff
@@ -63,11 +64,35 @@ const fetcher = async (
   return json?.data?.results || [];
 };
 
+const reportPossibleMatch = async (
+  payload: { lastReportedId: string | undefined; possibleMatchId: number },
+  getServerApi: Function
+): Promise<void> => {
+  const { lastReportedId, possibleMatchId } = payload;
+  const memorizedDogId: string | null = decryptData("lastReportedDogId");
+
+  if (!lastReportedId && !memorizedDogId)
+    return console.error("No memorized dog id in both URL and localStorage");
+
+  if (
+    lastReportedId &&
+    (!memorizedDogId || (memorizedDogId && lastReportedId !== memorizedDogId))
+  )
```

Comment:
I would recommend wrapping every call that includes i/o operation with the LS in try/catch decleration for a proper handling,  at the encryo util file instead of console.error and returning null, throw proper error when failed.